relayer/crates/starknet-integration-tests/src/tests/clear_packets.rs

```diff
@@ -0,0 +1,599 @@
+            .bootstrap_connection(&conn_init_option)
+            .await?;
+
+        info!("starknet_connection_id: {:?}", starknet_connection_id);
+        info!("cosmos_connection_id: {:?}", cosmos_connection_id);
+
+        // channel handshake
+
+        let ics20_port = IbcPortId::transfer();
+
+        {
+            // register the ICS20 contract with the IBC core contract
+
+            let register_app = MsgRegisterApp {
+                port_id: ics20_port.clone(),
+                contract_address: ics20_contract_address,
+            };
+
+            let register_call_data = cairo_encoding.encode(&register_app)?;
+
+            let call = Call {
+                to: ibc_core_address,
+                selector: selector!("bind_port_id"),
+                calldata: register_call_data,
+            };
+
+            let message = StarknetMessage::new(call);
+
+            let response = starknet_chain.send_message(message).await?;
+
+            info!("register ics20 response: {:?}", response);
+        }
+
+        let init_channel_options = CosmosInitChannelOptions::new(starknet_connection_id);
+
+        let (starknet_channel_id, cosmos_channel_id) = starknet_to_cosmos_relay
+            .bootstrap_channel(
+                &ics20_port.clone(),
+                &ics20_port.clone(),
+                &init_channel_options,
+            )
+            .await?;
+
+        info!("starknet_channel_id: {:?}", starknet_channel_id);
+        info!("cosmos_channel_id: {:?}", cosmos_channel_id);
+
+        // First ics20 transfer to Cosmos
+
+        let wallet_cosmos_a = &cosmos_chain_driver.user_wallet_a;
+        let address_cosmos_a = &wallet_cosmos_a.address;
+        let wallet_starknet_b = &starknet_chain_driver.user_wallet_b;
+        let address_starknet_b = &wallet_starknet_b.account_address;
+        let transfer_quantity = 1_000u128;
+        let denom_cosmos = &cosmos_chain_driver.genesis_config.transfer_denom;
+
+        let balance_cosmos_a_step_0 = cosmos_chain
+            .query_balance(address_cosmos_a, denom_cosmos)
+            .await?;
+
+        info!(
+            "cosmos balance before transfer: {}",
+            balance_cosmos_a_step_0
+        );
+
+        let commitment_sequences = vec!(Sequence::from(1));
+
+        let pending_before = <StarknetChain as CanQueryUnreceivedPacketSequences<CosmosChain>>::query_unreceived_packet_sequences(
+            starknet_chain, &starknet_channel_id,
+            &IbcPortId::transfer(),
+            commitment_sequences.as_slice()
+        ).await?;
+
+        let packet = <CosmosChain as CanIbcTransferToken<StarknetChain>>::ibc_transfer_token(
+            cosmos_chain,
+            &cosmos_channel_id,
+            &IbcPortId::transfer(),
+            wallet_cosmos_a,
+            address_starknet_b,
+            &Amount::new(transfer_quantity, denom_cosmos.clone()),
+            &None,
+        )
+        .await?;
+
+        assert_eq!(pending_before, commitment_sequences);
```

Comment:
We can't add these asserts in `ics20.rs` as we are using auto-relaying, right?

Reply:
Exactly, and as mentioned at the top, this test will undergo changes when adding the query pending acks. As this will require doing partial relaying manually.